ipn/ipnlocal/taildrop.go

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package ipnlocal
+
+import (
+	"slices"
+	"strings"
+
+	"tailscale.com/ipn"
+)
+
+func (b *LocalBackend) UpdateOutgoingFiles(updates map[string]ipn.OutgoingFile) {
+	b.mu.Lock()
+	if b.outgoingFiles == nil {
+		b.outgoingFiles = make(map[string]*ipn.OutgoingFile, len(updates))
+	}
+	for id, file := range updates {
+		b.outgoingFiles[id] = &file
```

Comment:
It's too soon to depends on this IMO.  Use ptr.To instead?

Reply:
Sorry, I don't understand. What is this depending on?

Reply:
Oh, I thought you were being clever and intentionally taking advantage of the Go 1.22 language change: https://go.dev/doc/go1.22#language

It's subtle enough (few enough people understand the old rules, much less the new rules) that I'd prefer that code be explicit about what it's trying to do.

But the bigger questionable part is why you have some parts of the code using a map[K]V and some using map[K]*V. I'd pick one or the other. And then what's the goal here? Just a maps.Clone?

Reply:
> But the bigger questionable part is why you have some parts of the code using a map[K]V

Just avoiding a one last copy since the file is going to be heap allocated anyway. +1 to being consistent though, will update.

Reply:
Okay, I pushed the referencing further up the stack. I still need to use a non-pointer OutgoingFile on a channel to avoid concurrency issues inside of `serveFilePut`, but other than that we're now just using the pointer version.